src/modules/Dropdown/Dropdown.js

```diff
@@ -686,10 +702,12 @@ export default class Dropdown extends Component {
       search && searchQuery && 'filtered'
     )
     let _text = placeholder
-    if (text && !searchQuery) {
-      _text = text
-    } else if (searchQuery) {
+    if (searchQuery) {
       _text = null
+    } else if (text) {
+      _text = text
+    } else if (open && !multiple) {
+      _text = this.getSelectedItemText()
```

Comment:
This line is the same as 712 below:

```jsx
/* 710 */ _text = this.getSelectedItemText()
/* 712 */ _text = _.get(this.getItemByValue(value), 'text')
```

Suggest removing the `getSelectedItemText` method, since it is only used once.  It is best to leave the code inline vs adding indirection.  Then, combine the if/else clauses:

```js
} else if (hasValue || open && !multiple) {
  _text = _.get(this.getItemByValue(value), 'text')
}
```